pkg/apis/stork/v1alpha1/types.go

```diff
@@ -212,3 +217,193 @@ type MigrationList struct {
+	Network NetworkSpec `json:"network"`
+	// Storage details of storage used by the driver
+	Storage StorageSpec `json:"storage"`
+	// Geo is topology information for the node
+	Geo Geography `json:"geography"`
+	// Env is a list of environment variables used by the driver
+	Env []core.EnvVar `json:"env"`
+	// RuntimeOpts is a map of options with extra configs for storage driver
+	RuntimeOpts map[string]string `json:"runtimeOptions"`
+}
+
+// NodeSelector let's the user select a node or group of nodes based on either
+// the NodeName or the node LabelSelector. If NodeName is specified then,
+// LabelSelector is ignored as that is more precise, even though it does not
+// match any node names.
+type NodeSelector struct {
+	// NodeName is the name of Kubernetes node that it to be selected
+	NodeName string `json:"nodeName"`
+	// LabelSelector is label query over all the nodes in the cluster
+	LabelSelector meta.LabelSelector `json:"labelSelector"`
+}
+
+// KvdbSpec contains the details to access kvdb
+type KvdbSpec struct {
+	// Internal flag indicates whether to use internal kvdb or an external one
+	Internal bool `json:"internal"`
+	// Endpoints to access the kvdb
+	Endpoints []string `json:"endpoints"`
+	// BasicAuthSecret is name of kubernetes secret containing the username
+	// and password for basic auth
+	BasicAuthSecret string `json:"accessSecret"`
+	// CertificateSecret is name of the kubernetes secret containing the
+	// cert files required for kvdb authentication
+	CertificateSecret string `json:"certificateSecret"`
+	// ACLTokenSecret is name of the kubernetes secret containing the
+	// ACL token for kvdb authentication
+	ACLTokenSecret string `json:"aclTokenSecret"`
+}
+
+// NetworkSpec contains network information
+type NetworkSpec struct {
+	// DataInterface is the network interface used by driver for data traffic
+	DataInterface string `json:"dataInterface"`
+	// MgmtInterface is the network interface used by driver for mgmt traffic
+	MgmtInterface string `json:"mgmtInterface"`
+}
+
+// StorageSpec details of storage used by the driver
+type StorageSpec struct {
+	// UseAll use all available, unformatted, unpartioned devices.
+	// This will be ignored if Devices is not empty.
+	UseAll bool `json:"useAll"`
+	// UseAllWithPartitions use all available unformatted devices
+	// including partitions. This will be ignored if Devices is not empty.
+	UseAllWithPartitions bool `json:"useAllWithPartitions"`
+	// Devices list of devices to be used by storage driver
+	Devices []string `json:"devices"`
+	// StorageMdDevices metadata device for storage pools
+	StorageMdDevices []string `json:"storageMetadataDevices"`
+	// JournalDevice device for journaling
+	JournalDevice string `json:"journalDevice"`
+	// SystemMdDevice device that will be used to store system metadata
+	SystemMdDevice string `json:"systemMetadataDevice"`
+	// DataStorageType type of data storage used
+	DataStorageType string `json:"dataStorageType"`
```

Comment:
Needs more info on what this means.

Reply:
@jvinod Do we need to expose this field to users? If it's for internal testing, can we just put it in RuntimeOpts and add necessary flags in oci mon?